src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -142,4 +144,10 @@ public void removeUserBook(@PathVariable(name = "idBook") Long idBook,
         user.removeBook(book);
         userRepository.save(user);
     }
+
+    @RequestMapping(value = "/username")
+    public String getUserName(Authentication authentication) {
+        Optional<User> name = (Optional<User>) authentication.getPrincipal();
+        return name.get().getUserName();
```

Comment:
You should check if there is a User object or not, even if it looks like there will always be.